src/tables.rs

```diff
@@ -192,6 +192,211 @@ impl BlockSize {
+        },
+        R4x8 => TxfmInfo {
+            w: 1,
+            h: 2,
+            lw: 0,
+            lh: 1,
+            min: 0,
+            max: 1,
+            sub: S4x4,
+            ctx: 1,
+        },
+        R8x4 => TxfmInfo {
+            w: 2,
+            h: 1,
+            lw: 1,
+            lh: 0,
+            min: 0,
+            max: 1,
+            sub: S4x4,
+            ctx: 1,
+        },
+        R8x16 => TxfmInfo {
+            w: 2,
+            h: 4,
+            lw: 1,
+            lh: 2,
+            min: 1,
+            max: 2,
+            sub: S8x8,
+            ctx: 2,
+        },
+        R16x8 => TxfmInfo {
+            w: 4,
+            h: 2,
+            lw: 2,
+            lh: 1,
+            min: 1,
+            max: 2,
+            sub: S8x8,
+            ctx: 2,
+        },
+        R16x32 => TxfmInfo {
+            w: 4,
+            h: 8,
+            lw: 2,
+            lh: 3,
+            min: 2,
+            max: 3,
+            sub: S16x16,
+            ctx: 3,
+        },
+        R32x16 => TxfmInfo {
+            w: 8,
+            h: 4,
+            lw: 3,
+            lh: 2,
+            min: 2,
+            max: 3,
+            sub: S16x16,
+            ctx: 3,
+        },
+        R32x64 => TxfmInfo {
+            w: 8,
+            h: 16,
+            lw: 3,
+            lh: 4,
+            min: 3,
+            max: 4,
+            sub: S32x32,
+            ctx: 4,
+        },
+        R64x32 => TxfmInfo {
+            w: 16,
+            h: 8,
+            lw: 4,
+            lh: 3,
+            min: 3,
+            max: 4,
+            sub: S32x32,
+            ctx: 4,
+        },
+        R4x16 => TxfmInfo {
+            w: 1,
+            h: 4,
+            lw: 0,
+            lh: 2,
+            min: 0,
+            max: 2,
+            sub: R4x8,
+            ctx: 1,
+        },
+        R16x4 => TxfmInfo {
+            w: 4,
+            h: 1,
+            lw: 2,
+            lh: 0,
+            min: 0,
+            max: 2,
+            sub: R8x4,
+            ctx: 1,
+        },
+        R8x32 => TxfmInfo {
+            w: 2,
+            h: 8,
+            lw: 1,
+            lh: 3,
+            min: 1,
+            max: 3,
+            sub: R8x16,
+            ctx: 2,
+        },
+        R32x8 => TxfmInfo {
+            w: 8,
+            h: 2,
+            lw: 3,
+            lh: 1,
+            min: 1,
+            max: 3,
+            sub: R16x8,
+            ctx: 2,
+        },
+        R16x64 => TxfmInfo {
+            w: 4,
+            h: 16,
+            lw: 2,
+            lh: 4,
+            min: 2,
+            max: 4,
+            sub: R16x32,
+            ctx: 3,
+        },
+        R64x16 => TxfmInfo {
+            w: 16,
+            h: 4,
+            lw: 4,
+            lh: 2,
+            min: 2,
+            max: 4,
+            sub: R32x16,
+            ctx: 3,
+        },
+    }
+}
+
 pub static dav1d_txfm_dimensions: [TxfmInfo; TxfmSize::COUNT] = {
```

Comment:
Does this need to be a static? Can it be const so we don't have to duplicate it in the const switch above? I don't see anywhere that really relies on being able to take the address of these elements (i.e. they don't seem to be used from ASM?)

Reply:
I hadn't seen a need for it t remain a static, but preferred to leave this out of the experiment to have your opinion about the const generics first.

Reply:
You could also use `enum_map!` to use the above `fn` to generate this table (and make the key typed).

Reply:
I did try `enum_map!` but then my measurements seemed slightly worse. It also required implement `DefaultValue` for `TxfmInfo` that I'm not sure can be correct, but then again it won't be used by the enum map so it may be fine?

Since the measurements seemed slightly worsened (and I haven't had the time t check the assembly), but still within noise most of the time, I didn't add this change to this PR. I can definitely push it to another branch if you'd like to test it yourself.

I also tried completely removing the `static` and using the (modified) `const fn` for the runtime values outside of `decode_coefs`, but that also seemed to yield worse numbers, so I've left it out of this PR. I did push a commit removing the duplication by using the new function though.